### ISSUE TITLE
[Floating UI] Nested floaters have incorrect initial positioning when used with React Portals

### ISSUE THREAD

**Issue by valtism** [floating-ui/floating-ui#1531]
Reproduction: https://stackblitz.com/edit/floating-ui-nested-portals?file=src/App.js

Notice how the Floating div is initially placed at the bottom of the window, only updating to the correct position after a scroll event.

This only happens when using a portal to place the floating element at the modal-root of the DOM tree, something that we need to do to deal with popping out of scrolling containers.

**Comment by valtism** [comment]
Actually, after playing with this a little, it seems like using a portal may not be necessary. I thought (maybe from popperjs, or perhaps a holdover from doing this manually) that scrolling container elements would not allow floating element to break out of them, but that doesn't seem to be the case here. 

Do you suggest not using portals at all, or is there some use-case I am missing here?

**Comment by valtism** [comment]
Ah, that's why I wasn't experiencing it. Thanks for the info on clipping containers.

As for the inside-out rendering, is there some way to avoid this? I'm guessing that this happens anyway with nested floating components, but why does it position it correctly without the portal? 

Is there a suggested way to work around this behaviour with portals? I think I can patch the issue with a useEffect like

```js
useEffect(() => {
    setTimeout(() => {
      update()
    },10)
  })
  ```
  
but that seems a little hacky.

**Comment by valtism** [comment]
I'm not sure if this approach would work, since in the example `<Float>` was a wrapper component and has no knowledge of what children it renders. Updating the wrapper component does not trigger an update to the nested children afaik.

I was able to work around this issue because I created a component that uses context to pass values into a nested reference and floating compound components, and this seems to trigger a re-render at the right time to avoid the issue, but I can see this still being a problem if you are taking a more straightforward approach.
